### PR TITLE
Move CLI parsing

### LINKED ISSUE #201: Move parsing of command line before parser creation
Some external users may need to know when Kokkos (or Tpetra or VT) is used before creating the parser.

### PR DESCRIPTION
**Pull request by uhetmaniuk**
Delete typedef `MaterialFactoryType`.

Closes #201

### CODE REVIEW

**Comment by uhetmaniuk** [comment]
@japlews 
I have tried to address the concern that you raised when the Parser variable depends on the CLI flag.
I have also removed the typedef `MaterialFactoryType`.
I believe the local section in `int main()` should address the deletion of all Kokkos variables before the finalize.
Let me know if some tests are still failing.

File: src/nimble.cc [hunk 1]
```diff
@@ -45,37 +45,53 @@
...
   //
-  // note: If the command line does not specify any parameter '--use-kokkos',
-  // '--use-tpetra', or '--use-vt', we could choose to specify one of them with
-  // either myParser.SetToUseKokkos() or myParser.SetToUseTpetra() or
-  // myParser.SetToUseVT()
+  // Quick parsing of command line to check whether Kokkos, Tpetra, or VT is used.
   //
-  nimble::NimbleInitializeAndGetInput(argc, argv, myParser);
+  nimble::EnvironmentFlags myFlags;
+  details::parseCommandLine(argc, argv, myFlags);
```
Comment (japlews) [inline src/nimble.cc]: Can you make this accessible from the nimble_main.h header? Otherwise, all looks good!

**Review by japlews — CHANGES_REQUESTED** [review]
Minor fix needed to make this work.

**Review by nmm0 — COMMENTED** [review]
Thumbs up from me -- just contingent on the minor change @japlews requested. One thing we should definitely consider in the future is using a library for CLI parsing. There are definitely a lot of libraries that do it better than we do. That's pretty low priority though.

**Comment by japlews** [comment]
Totally agree with @nmm0, it would be very easy to switch over to YAML and use a simple open source library for that.

**Review by japlews — APPROVED** [review]
Builds for me and tests pass.

**Pull request merged**